app/controllers/spaces_controller.rb

```diff
@@ -5,11 +5,25 @@ class SpacesController < ApplicationController
 
   #GET /spaces
   #GET /spaces.json
+  #TODO(matthew): params['strings'] or params[:symbols]?
   def index
     if params['user_id']
       @user = User.find(params['user_id'])
-      @spaces = @user.spaces.visible_by(current_user)
+      if current_user && @user.id == current_user.id
+        @spaces = @user.spaces
+      else
+        @spaces = @user.spaces.is_public
+      end
+    elsif params['organization_id']
+      if current_user && current_user.member_of?(params['organization_id'])
+        @spaces = Organization.find(params['organization_id']).spaces
+      else
+        head :unauthorized
```

Comment:
this section could be cleaned up (not for this PR).  I guess ``member_of?`` isn't very specific, but it's more low-level than I would want in a controller.